internal/dataplane/kongstate/customentity.go

```diff
@@ -89,3 +97,61 @@ func ExtractEntityFieldDefinitions(schema kong.Schema) EntitySchema {
 	}
 	return retSchema
 }
+
+// IsKnownEntityType returns true if the entities of the type are "standard" and processed elsewhere in KIC.
+func IsKnownEntityType(entityType string) bool {
+	// Types of standard Kong entities that are processed elsewhere in KIC.
+	// So the entities cannot be specified via KongCustomEntity types.
+	knownEntities := map[kong.EntityType]struct{}{
+		kong.EntityTypeServices:       {},
+		kong.EntityTypeRoutes:         {},
+		kong.EntityTypeUpstreams:      {},
+		kong.EntityTypeTargets:        {},
+		kong.EntityTypeConsumers:      {},
+		kong.EntityTypeConsumerGroups: {},
+		kong.EntityTypePlugins:        {},
+	}
+	_, ok := knownEntities[kong.EntityType(entityType)]
+	return ok
+}
```

Comment:
Food for thought on the optimization topic:

This approach:

```
go test -run NONE -benchmem -bench . -count 3 ./internal/dataplane/kongstate
goos: darwin
goarch: arm64
pkg: github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/kongstate
BenchmarkIsKnownEntityType-10            5477181               213.7 ns/op             0 B/op          0 allocs/op
BenchmarkIsKnownEntityType-10            5584881               208.5 ns/op             0 B/op          0 allocs/op
BenchmarkIsKnownEntityType-10            5731390               210.6 ns/op             0 B/op          0 allocs/op
PASS
ok      github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/kongstate   4.572s
```

vs

```
func IsKnownEntityType(entityType string) bool {
	switch entityType {
	case
		string(kong.EntityTypeServices),
		string(kong.EntityTypeRoutes),
		string(kong.EntityTypeUpstreams),
		string(kong.EntityTypeTargets),
		string(kong.EntityTypeConsumers),
		string(kong.EntityTypeConsumerGroups),
		string(kong.EntityTypePlugins):
		return true
	default:
		return false
	}
}
```

```
go test -run NONE -benchmem -bench . -count 3 ./internal/dataplane/kongstate
goos: darwin
goarch: arm64
pkg: github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/kongstate
BenchmarkIsKnownEntityType-10            8790510               126.1 ns/op             0 B/op          0 allocs/op
BenchmarkIsKnownEntityType-10            9418054               125.8 ns/op             0 B/op          0 allocs/op
BenchmarkIsKnownEntityType-10            9251524               127.4 ns/op             0 B/op          0 allocs/op
PASS
ok      github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/kongstate   4.299s
```


```
func BenchmarkIsKnownEntityType(b *testing.B) {
	for i := 0; i < b.N; i++ {
		require.True(b, IsKnownEntityType(string(kong.EntityTypeServices)))
	}
}
```

Reply:
Updated.